src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2821,6 +2825,50 @@ SqlTaskList(Job *job)
 }
 
 
+/*
+ * RelabelTypeMutator walks over the tree and converts RelabelType's
+ * into CollationExpr's. With that, we will be able to pushdown COLLATE's.
+ */
+static Node *
+RelabelTypeMutator(Node *originalNode)
+{
+	Node *newNode = NULL;
+	if (originalNode == NULL || !IsA((Query *) originalNode, Query))
+	{
+		return NULL;
+	}
+
+	if (IsA(originalNode, RelabelType))
+	{
+		RelabelType *originalRelabelType = (RelabelType *) originalNode;
```

Comment:
instead of `*originalRelabelType` maybe just `*relabelType` and
`*newCollateExpr` -> `*collateExpr`